proposals/0000-quic-turbine.md

```diff
@@ -0,0 +1,95 @@
+---
+simd: '0053'
+title: Block Propagation Using QUIC Protocol
+authors:
+  - Solana Labs
+category: Standard
+type: Networking
+status: Draft
+created: 2023-05-31
+feature: (fill in with feature tracking issues once accepted)
+---
+
+## Summary
+
+Use QUIC protocol (instead of current UDP implementation) to propagate blocks
+across cluster.
+
+## Motivation
+
+* Compared to UDP, QUIC provides better features to preserve QoS.
+* Verifying Turbine propagation path (see
+  [solana/#28384](https://github.com/solana-labs/solana/issues/28384) and
+  [solana/#20969](https://github.com/solana-labs/solana/issues/20969)) requires
+  some mechanism to prevent IP address spoofing, which is readily available
+  with QUIC.
+
+## Alternatives Considered
+
+An alternative would be to keep current UDP implementation but,
+
+* Implement [ping/pong
+  protocol](https://github.com/solana-labs/solana/blob/2fc1dc1bf/gossip/src/ping_pong.rs)
+  between nodes to verify IP addresses before retransmitting any shreds to the
+  recipient.
+* Each node also signs and attaches its signature to each shred it is
+  retransmitting.
+
+This will cost 64 bytes of each shred payload to embed retransmitter's
+signature and an extra signature verification overhead for each shred at the
+recipient node.
+
+There are ways to improve above but it gradually approaches to re-inventing
+QUIC which defeats the point of not using QUIC to begin with.
+
+## New Terminology
+
+TVU QUIC socket address refers to the port which validator nodes will receive
+shreds using QUIC protocol.
+
+## Detailed Design
+
+Once the cluster has upgraded to the new
+[`ContactInfo`](https://github.com/solana-labs/solana/blob/2fc1dc1bf/gossip/src/contact_info.rs#L68-L85)
+nodes may explicitly specify a socket address for TVU QUIC connections.
+Until then, similar to QUIC migration for TPU, we will use the port at
+[`QUIC_PORT_OFFSET`](https://github.com/solana-labs/solana/blob/2fc1dc1bf/sdk/src/quic.rs#L4)
+from the TVU socket for QUIC connections.
+
+For the initial implementation in the Solana Labs client we will use the same
+constructs as TPU QUIC implementation so the specs are the same. See [TPU/QUIC
+Protocol v1](https://github.com/solana-foundation/specs/blob/42f2058b7/p2p/tpu.md#tpuquic-protocol-v1).
```

Comment:
stake weighted stream / connection limit might become an issue with this use-case

Reply:
Turbine random shuffle is also stake weighted; so I would expect some positive correlation between stake and who I receive shreds from.
It might be the case that the current cache-eviction policy for TPU connections wouldn't work well for turbine, in which case we can plug in a different eviction policy; but I guess that will still need to favor higher staked nodes somehow.